stable/aws-for-fluent-bit/README.md

```diff
@@ -96,15 +116,80 @@ helm delete aws-for-fluent-bit --namespace kube-system
-| `tolerations` | Optional deployment tolerations | `[]` |
-| `nodeSelector` | Node labels for pod assignment | `{}` |
-| `annotations` | Optional pod annotations | `{}` |
-| `volumes` | Volumes for the pods, provide as a list of volume objects (see values.yaml) |  volumes for /var/log and /var/lib/docker/containers are present, along with a fluentbit config volume |
-| `volumeMounts` | Volume mounts for the pods, provided as a list of volumeMount objects (see values.yaml) | volumes for /var/log and /var/lib/docker/containers are mounted, along with a fluentbit config volume |
-| `dnsPolicy` | Optional dnsPolicy | `ClusterFirst` |
-| `hostNetwork` | If true, use hostNetwork | `false` |
+| `s3.enabled` | Whether this plugin should be enabled or not, [details](https://docs.fluentbit.io/manual/pipeline/outputs/s3) | `false`
+| `s3.match` | The log filter. | `"*"`
+| `s3.bucket` | S3 Bucket name. |
+| `s3.region` | The AWS region of your S3 bucket. | `"us-east-1"`
+| `s3.jsonDateKey` | Specify the name of the time key in the output record. To disable the time key just set the value to false. | `"date"`
+| `s3.jsonDateFormat` | Specify the format of the date. Supported formats are double, epoch, iso8601 (eg: 2018-05-30T09:39:52.000681Z) and java_sql_timestamp (eg: 2018-05-30 09:39:52.000681). | `"iso8601"`
+| `s3.totalFileSize` | Specifies the size of files in S3. Maximum size is 50G, minimim is 1M. | `"100M"`
+| `s3.uploadChunkSize` | The size of each 'part' for multipart uploads. Max: 50M | `"6M"`
+| `s3.uploadTimeout` | Whenever this amount of time has elapsed, Fluent Bit will complete an upload and create a new file in S3. For example, set this value to 60m and you will get a new file every hour. | `"10m"`
+| `s3.storeDir` | Directory to locally buffer data before sending. When multipart uploads are used, data will only be buffered until the `upload_chunk_size` is reached. S3 will also store metadata about in progress multipart uploads in this directory; this allows pending uploads to be completed even if Fluent Bit stops and restarts. It will also store the current $INDEX value if enabled in the S3 key format so that the $INDEX can keep incrementing from its previous value after Fluent Bit restarts. | `"/tmp/fluent-bit/s3"`
+| `s3.storeDirLimitSize` | The size of the limitation for disk usage in S3. Limit the amount of s3 buffers in the `store_dir` to limit disk usage. Note: Use `store_dir_limit_size` instead of `storage.total_limit_size` which can be used to other plugins, because S3 has its own buffering system. | `0`
+| `s3.s3KeyFormat` | Format string for keys in S3. This option supports a UUID, strftime time formatters, a syntax for selecting parts of the Fluent log tag using a syntax inspired by the rewrite_tag filter. Add $TAG in the format string to insert the full log tag; add $TAG[0] to insert the first part of the tag in the s3 key. | `"/pod-logs/$TAG/%Y-%m-%d/%H-%M-%S"`
```

Comment:
For the most part, the readme is just copying the docs. Except this one, where its shorter and has less info. I'd prefer if we either copy the full doc for s3_key_format, or, better: link to it. And your should description should note all of the special things that can be put in the key, `$INDEX`, `$TAG`, `$TAG[0]`, `$UUID` and then say see S3 docs for deep explanation.

Reply:
done